### PR TITLE
hidapi: fix cross build

### PR DESCRIPTION
**Pull request by FRidh**
###### Motivation for this change

Running ofborg to assess the amount of rebuilds. Might need to make this a cross-only patch.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/hidapi/default.nix [hunk 1]
```diff
@@ -1,5 +1,7 @@
 { stdenv, fetchFromGitHub, autoreconfHook, pkgconfig, udev, libusb1
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/hidapi/default.nix:1]:
```suggestion
{ stdenv, fetchFromGitHub, autoreconfHook, pkg-config, udev, libusb1
```

File: pkgs/development/libraries/hidapi/default.nix [hunk 2]
```diff
@@ -12,7 +14,11 @@ stdenv.mkDerivation rec {
     sha256 = "1nr4z4b10vpbh3ss525r7spz4i43zim2ba5qzfl15dgdxshxxivb";
   };
 
-  nativeBuildInputs = [ autoreconfHook pkgconfig ];
+  nativeBuildInputs = [
+    autoreconfHook
+    pkgconfig
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/hidapi/default.nix:19]:
```suggestion
    pkg-config
```

File: pkgs/development/libraries/hidapi/default.nix [hunk 3]
```diff
@@ -12,7 +14,11 @@ stdenv.mkDerivation rec {
...
+    gnum4
+  ];
 
   buildInputs = [ ]
     ++ stdenv.lib.optionals stdenv.isLinux [ libusb1 udev ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/hidapi/default.nix:24]:
```suggestion
  buildInputs = stdenv.lib.optionals stdenv.isLinux [ libusb1 udev ];
```

**Comment by FRidh** [comment]
ok, this needs to be added conditionally

**Comment by FRidh** [comment]
pushed f6514239ee9b0739a6e884ba28e9302c0102f8c0 to staging-next

**Comment by alyssais** [comment]
Thanks for the fix.

**Comment by flokli** [comment]
I pushed a cleanup PR on top, making this not only conditional on cross in https://github.com/NixOS/nixpkgs/pull/108044, and addressing the PR feedback.

**Pull request closed without merging**